cartographer/mapping_2d/sparse_pose_graph.cc

```diff
@@ -106,6 +106,8 @@ void SparsePoseGraph::AddScan(
   });
   trajectory_connectivity_.Add(submaps);
 
+  // if the last (i.e. latest) insertion_submap is not in the index
+  // it means we have started a new submap...
```

Comment:
To me it was weird, that there can be a submap, that scans are being inserted to, which is not even in the index yet.
How about 
```
// SPG learns about new submaps from unseen indices in 'insertion_submaps'
```